src/navigable-group/__tests__/navigable-group.test.tsx

```diff
@@ -0,0 +1,310 @@
+        const { wrapper } = renderNavigableGroup({ navigationDirection: 'vertical' });
+
+        const button1 = wrapper.findAllButtons()[0];
+        const button2 = wrapper.findAllButtons()[1];
+
+        button2.focus();
+        button2.keydown(KeyCode.up);
+
+        expect(button1.getElement()).toHaveFocus();
+      });
+
+      test('does not navigate with ArrowLeft/ArrowRight keys in vertical mode', () => {
+        const { wrapper } = renderNavigableGroup({ navigationDirection: 'vertical' });
+
+        const button2 = wrapper.findAllButtons()[1];
+
+        button2.focus();
+        button2.keydown(KeyCode.left);
+        expect(button2.getElement()).toHaveFocus();
+        button2.keydown(KeyCode.right);
+        expect(button2.getElement()).toHaveFocus();
+      });
+    });
+
+    describe('Both directions', () => {
+      test('navigates with all arrow keys when direction is "both"', () => {
+        const { wrapper } = renderNavigableGroup({ navigationDirection: 'both' });
+
+        const button1 = wrapper.findAllButtons()[0];
+        const button2 = wrapper.findAllButtons()[1];
+        const button3 = wrapper.findAllButtons()[2];
+
+        button2.focus();
+        button2.keydown(KeyCode.up);
+        expect(button1.getElement()).toHaveFocus();
+        button1.keydown(KeyCode.down);
+        expect(button2.getElement()).toHaveFocus();
+        button2.keydown(KeyCode.right);
+        expect(button3.getElement()).toHaveFocus();
+        button3.keydown(KeyCode.left);
+        expect(button2.getElement()).toHaveFocus();
+      });
+    });
+
+    describe('Home and End keys', () => {
+      test('navigates to last item with end key', () => {
+        const { wrapper } = renderNavigableGroup();
+
+        const button1 = wrapper.findAllButtons()[0];
+        const button3 = wrapper.findAllButtons()[2];
+
+        button1.focus();
+        button1.keydown(KeyCode.end);
+        expect(button3.getElement()).toHaveFocus();
+      });
+
+      test('navigates to first item with home key', () => {
+        const { wrapper } = renderNavigableGroup();
+
+        const button1 = wrapper.findAllButtons()[0];
+        const button3 = wrapper.findAllButtons()[2];
+
+        button3.focus();
+        button3.keydown(KeyCode.home);
+        expect(button1.getElement()).toHaveFocus();
+      });
+    });
+
+    test('loops focus', () => {
+      const { wrapper } = renderNavigableGroup();
+
+      const button1 = wrapper.findAllButtons()[0];
+      const button3 = wrapper.findAllButtons()[2];
+
+      button3.focus();
+      button3.keydown(KeyCode.right);
+      expect(button1.getElement()).toHaveFocus();
+      button1.keydown(KeyCode.left);
+      expect(button3.getElement()).toHaveFocus();
+    });
+
+    test('works with custom getItemKey function', () => {
+      const { container } = render(
+        <NavigableGroup getItemKey={element => element.getAttribute('data-custom-id') || element.id}>
```

Comment:
Nitpick: I think the `|| element.id` default could be removed because it's not used.